UnleashClient/constraints/Constraint.py

```diff
@@ -139,8 +144,16 @@ def check_date_operators(self, context_value: Union[datetime, str]) -> bool:
 
         try:
             parsed_date = parse(self.value)
+
+            # If parsed date is timezone-naive, assume it is UTC
+            if parsed_date.tzinfo is None:
+                parsed_date = parsed_date.replace(tzinfo=timezone.utc)
+
             if isinstance(context_value, str):
                 context_date = parse(context_value)
+                # If parsed date is timezone-naive, assume it is UTC
+                if context_date.tzinfo is None:
```

Comment:
I'm not sure this is the right thing to do here. I'd like both a a `Date` and a `str` to go through the same code path and right now one is raising an error and the other silently changing the underlying data